### ISSUE TITLE
Replace JDK 11 and JavaFX 16 with JDK 17 and JavaFX 17

### ISSUE THREAD

**Issue by benfry** [benfry/processing4#285]
Need to switch to the latest LTS versions of the JDK and JavaFX. Just requires a bit more hacking and testing to get everything in place. 

Got as far as the following when launching an app on macOS:

```
2021-10-04 19:18:16.119 Processing[36142:1724471] int launch(char *, int, char **) Error launching JVM Runtime (jdk-17+35) Relative Path: '/Users/fry/coconut/processing4/build/macosx/work/Processing.app/Contents/PlugIns/jdk-17+35' (dylib: (null))
  error: Unable to load Java Runtime Environment.
```

This may be something simple, or may just need an `appbundler` update (we're not using the latest source). But there are problem other quirks, so opting not to make the switch for beta 2, which will be focused on stability fixes.

**Comment by benfry** [comment]
As of https://github.com/processing/processing4/commit/f2e66037b529a32a1f58d9fac3f560b34ec0e0f9 it's possible to switch to Java 17 by commenting out and un-commenting two sections in `build.xml`.

Fixing this just requires some additional testing.

**Comment by benfry** [comment]
Resolved for 4.0 beta 3.
